src/Eloquent/DocumentModel.php

```diff
@@ -50,6 +50,7 @@
 use function strlen;
 use function var_export;
 
+/** @mixin Builder */
```

Comment:
I had to remove the `@mixin` annotation from the Model class (#2981) because the return types of the methods is not the same in the Builder and the Model. I must have made a mistake.